### PR TITLE
AAU, on the SDK, when I look at the API KEY to set, I don't see any KILI_USER_API_KEY anymore

### PR DESCRIPTION
**Pull request by FlorianLeRoyKili**
- [ ] I opened a merge request on `kili` on a branch with the same name than the branch of the current pull request and forced security tests.
- [ ] I opened a pull request on the Node playground if the changes are breaking.

### CODE REVIEW

File: kili/authentication.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class KiliAuth:
     """
 
     def __init__(self,
-                 api_key=os.getenv('KILI_USER_API_KEY'),
+                 api_key=os.getenv('KILI_API_KEY'),
```
Comment (PierreLeveau) [inline kili/authentication.py:37]: you can remove the default values for `api_key` and `api_endpoint`, as they are already passed by the client

File: recipes/counterfactual_data_augmentation.ipynb [hunk 2]
```diff
@@ -37,7 +37,7 @@
     "# !pip install kili # uncomment if you don't have kili installed already\n",
     "from kili.client import Kili\n",
     "\n",
-    "api_key = os.getenv('KILI_USER_API_KEY')\n",
+    "api_key = os.getenv('KILI_API_KEY')\n",
```
Comment (PierreLeveau) [inline recipes/counterfactual_data_augmentation.ipynb:40]: this is the default behavior of the Kili client, so you can remove the `api_key` arg in all the Kili client initializations (here and elsewhere)

File: recipes/create_project.ipynb [hunk 3]
```diff
@@ -140,7 +140,7 @@
     "# !pip install kili # uncomment if you don't have kili installed already\n",
     "from kili.client import Kili\n",
     "\n",
-    "api_key = os.getenv('KILI_USER_API_KEY')\n",
+    "api_key = os.getenv('KILI_API_KEY')\n",
```
Comment (PierreLeveau) [inline recipes/create_project.ipynb:143]: rm

File: recipes/create_project.ipynb [hunk 4]
```diff
@@ -140,7 +140,7 @@
...
     "api_endpoint = os.getenv('KILI_API_ENDPOINT') # If you use Kili SaaS, use the url 'https://cloud.kili-technology.com/api/label/v2/graphql'\n",
     "\n",
     "kili = Kili(api_key=api_key, api_endpoint=api_endpoint)"
```
Comment (PierreLeveau) [inline recipes/create_project.ipynb:146]: rm `api_key` argument

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
remove all the `api_key` arguments when the kili client is called (its default value is precisely the `KILI_API_KEY` value: https://github.com/kili-technology/kili-python-sdk/blob/08221296917e582cb12659834f74502653b19af9/kili/client.py#L58 )

**Review by theodu — APPROVED** [review]

File: recipes/create_project.ipynb (hunk 4 above)
Comment (theodu) [reply recipes/create_project.ipynb]: I like this message!

File: kili/exceptions.py [hunk 5]
```diff
@@ -12,6 +12,17 @@ def __str__(self):
         return f"Not found: '{self.name}'"
 
 
+class AuthenticationFailed(Exception):
+    """
+    Used when user is cannot authenticate to Kili endpoint
```
Comment (theodu) [inline kili/exceptions.py:17]: typo here

Comment (theodu) [reply kili/exceptions.py:17]: Used when the authentification fails

File: recipes/create_project.ipynb (hunk 4 above)
Comment (theodu) [reply recipes/create_project.ipynb]: in authentification.py, in KiliAuth, I see this code:
```

if api_key is None:
            message = 'You need to provide an API KEY to connect.' \
                ' Visit https://docs.kili-technology.com/reference/creating-an-api-key'
            warnings.warn(message, UserWarning)
```
Do you get this message ?
If api is None, we could change that to an error of type AuthenticationFailed too but with a message more customized to the situation

File: kili/exceptions.py [hunk 6]
```diff
@@ -12,6 +12,23 @@ def __str__(self):
         return f"Not found: '{self.name}'"
 
 
+class AuthenticationFailed(Exception):
```
Comment (PierreLeveau) [inline kili/exceptions.py:15]: nice

File: recipes/getting-started/getting_started-image_bounding_box_detection.ipynb [hunk 7]
```diff
@@ -77,7 +77,7 @@
         "# !pip install kili # uncomment if you don't have Kili installed already\n",
         "from kili.client import Kili\n",
         "\n",
-        "api_key = os.getenv('KILI_USER_API_KEY')\n",
+        "api_key = os.getenv('KILI_PI_KEY')\n",
```
Comment (PierreLeveau) [inline recipes/getting-started/getting_started-image_bounding_box_detection.ipynb:80]: KILI_API_KEY

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**